assets/javascripts/discourse/components/chat-composer.js

```diff
@@ -777,25 +698,8 @@ export default Component.extend(TextareaTextManipulation, ComposerUploadUppy, {
   },
 
   @action
-  cancelUploading(upload) {
-    this.appEvents.trigger(`${this.composerEventPrefix}:cancel-upload`, {
-      fileId: upload.id,
-    });
-    this.onValueChange?.(this.value, this.uploads, this.replyToMsg);
-  },
-
-  @action
-  removeUpload(upload) {
-    this.uploads.removeObject(upload);
-    this.onValueChange?.(this.value, this.uploads, this.replyToMsg);
-  },
-
-  @discourseComputed("composerDisabled", "uploads.[]", "inProgressUploads.[]")
-  showUploadsContainer(composerDisabled, uploads, inProgressUploads) {
-    if (composerDisabled) {
-      return false;
-    }
-
-    return uploads?.length > 0 || inProgressUploads?.length > 0;
+  uploadChanged(uploads) {
```

Comment:
its somewhat unexpected that method name is singular and it takes an array as arg

Reply:
Yes good catch, will fix!